### PR TITLE
Switched the documentation from Markdown to ReStructuredText

### LINKED ISSUE #929: Switch the documentation from Markdown to ReStructuredText
If FOSRestBundle maintainers agree on this, I'll send a pull request changing its documentation format from Markdown to reStructuredText.

In addition to being the official Symfony doc format, reST will allow us to quickly integrate this documentation on symfony.com bundles section. This will increase the exposure of the bundle and hopefully it will bring in more contributors.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This is the first step to improve the documentation of the bundle. As discussed, using `.rst` format is needed to integrate the bundle in symfony.com documentation section.

This fixes #929.

### CODE REVIEW

File: Resources/doc/1-setting_up_the_bundle.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+            // if you choose to use JMSSerializer, make sure that it is registered in your application
+
+            // new JMS\SerializerBundle\JMSSerializerBundle(),
+        );
+    }
+
+That was it!
+
+.. _``JMSSerializer``: https://github.com/schmittjoh/serializer
```
Comment (xabbuh) [inline Resources/doc/1-setting_up_the_bundle.rst]: use only one backtick here (also in the following two lines)

File: Resources/doc/7-manual-route-definition.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    public function getUsers() { /** */ }
+
+
+Result:
+
+===================  ======  ======  ====  ====================
+Name                 Method  Scheme  Host  Path
+===================  ======  ======  ====  ====================
+get_foo              GET      ANY    ANY   /users/foo.{_format}
+get_bar              GET      ANY    ANY   /users/bar.{_format}
```
Comment (xabbuh) [inline Resources/doc/7-manual-route-definition.rst]: Missing `===================  ======  ======  ====  ====================` at the end of the table and newline at the end of file.

File: Resources/doc/annotations-reference.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+.. code-block:: php
+
+    use FOS\RestBundle\Controller\Annotations\Route
+    /**
+    * @Route("", condition="context.getMethod() in ['GET', 'HEAD'] and request.headers.get('User-Agent') matches '/firefox/i'")
+    */
+
+.. _`@Route Symfony annotation`: http://symfony.com/doc/current/bundles/SensioFra
+meworkExtraBundle/annotations/routing.html
```
Comment (xabbuh) [inline Resources/doc/annotations-reference.rst]: wrong newline

File: Resources/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
+Getting Started With FOSRestBundle
+=====================================
+
+Installation
+------------
+
+Installation is a quick (I promise!) one-step process:
+
+1. :doc:`Setting up the bundle <1-setting_up_the_bundle.md>`
```
Comment (xabbuh) [inline Resources/doc/index.rst]: You need to remove the file extension here (same with the other references in this file).

**Comment by javiereguiluz** [comment]
@xabbuh thanks for the review! Everything is fixed now.

**Comment by xabbuh** [comment]
To actually being able to build the docs, you would also have to add the files to a (hidden) toctree and provide a config file for Sphinx.

**Comment by lsmith77** [comment]
I just released 1.5.0-RC3 .. so I merged this after wards so that we can tweak it as needed with the previous state locked in with that release.

**Pull request merged**